apps/bfd-pipeline/bfd-pipeline-rda-grpc/src/main/java/gov/cms/bfd/pipeline/rda/grpc/RDASink.java

```diff
@@ -0,0 +1,48 @@
+package gov.cms.bfd.pipeline.rda.grpc;
+
+import java.util.Collections;
+
+/**
+ * Interface for objects that process incoming objects. At least one of the methods must be
+ * implemented. All implementations are AutoCloseable since they will generally hold a database
+ * connection.
+ *
+ * @param <T> the type of objects processed
+ */
+public interface RDASink<T> extends AutoCloseable {
+  /**
+   * Write the object to the data store and return the number of objects successfully written.
+   *
+   * @param object single object to be written to the data store
+   * @return number of objects successfully processed
+   * @throws ProcessingException if the operation fails
+   */
+  default int writeObject(T object) throws ProcessingException {
+    return writeBatch(Collections.singleton(object));
```

Comment:
I'm a little confused about the dependencies between the two methods here, `writeObject` calls `writeBatch` and `writeBatch` calls `writeObject`?  Am I missing something?

Reply:
Both are default implementations of the methods in the interface.  An implementing class will override one of them and get the implementation of the other for free.

Reply:
neat; I'm guessing we'll want to override `writeBatch`, but we'll see how it plays out when we get the proto file from the RDA API team

Reply:
FWIW, this was also confusing to me until I opened up this resolved thread. Might be worth a comment about the intended use. I guess it sort of is with `at least one of the methods must be implemented`. But maybe making this quoted expectation clear:
> The idea is that a class that implements the interface will override just one of the two methods and get the other method implementation for free.